File: algorithms/M/Minimum Operations to Make the Array Increasing/Minimum Operations to Make the Array Increasing.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        count = 0
        for i in range(len(nums)-1):
            if nums[i] >= nums[i+1]:
                count += (nums[i]+1)-nums[i+1]
                nums[i+1] = nums[i]+1
        return count

File: algorithms/M/Minimum Operations to Make the Array Increasing/Minimum Operations to Make the Array Increasing.java
class Solution {
    public int minOperations(int[] nums) {
        if (nums.length <= 1) {
            return 0;
        }

        int count = 0;
        int num = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (num == nums[i]) {
                count++;
                num++;
            } else if (num > nums[i]) {
                num++;
                count += num - nums[i];
            } else {
                num = nums[i];
            }
        }
        
        return count;
    }
}
